src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/ManagedInstanceProtectorTests.ps1

```diff
@@ -0,0 +1,230 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+$mangedInstanceRg = "MlAndzic_RG"
+$managedInstanceName = "midemoinstancebc"
+$keyVaultName = "mitest-doNotDelete"
+$keyName = "mitest-key"
+$keyId = "https://mitest-donotdelete.vault.azure.net/keys/mitest-key/e97043455e14493ca05c3642549aedba"
+$keyVersion = "e97043455e14493ca05c3642549aedba"
+$tdeKeyName = $keyVaultName + "_" + $keyName + "_" + $keyVersion
+
+<#
+	.SYNOPSIS
+	Tests setting and getting encryption protector to service managed key 
+#>
+function Test-SetGetManagedInstanceEncryptionProtectorServiceManaged
```

Comment:
This is a lot of tests - each one of these test will add time to our CI build.  Instead could you have:
1 test for resourcegroupname, managedinstancename and name with type ServiceManaged
1 test for piping with type AzureKeyVault (as long as you test piping with object you don't need a separate test for input object)
1 test for resourceid with either
1 test for FailsWithoutKeyId

Reply:
Is it possible for a change to affect say a get cmdlet and not an add cmdlet for any given resource. If this is true then I would think that not having full coverage will cause bugs like these to go undetected in spite of unit tests. However, based on your recommendation I have limited the number of tests that run for CI, but, I would like to keep these for development testing.

Reply:
If you would like to keep all the tests that's fine (though I don't believe they are covering different scenarios) - can you combine all of it into one test? That will decrease the overall test time.

Reply:
what I am currently doing is following your guidance for CI tests (have all tests in a single test) and will leave the rest as skipped, so that CI is not affected. Having different test cases in different tests make it easier to root cause the issue.

Reply:
Okay sounds good to me.